github/src/main/kotlin/gropius/sync/github/Grabber.kt

```diff
@@ -0,0 +1,137 @@
+package gropius.sync.github
+
+import gropius.sync.github.generated.fragment.MetaData
+import gropius.sync.github.generated.fragment.PageInfoData
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.toList
+import org.bson.types.ObjectId
+import java.time.OffsetDateTime
+
+/**
+ * Requests data from github using steps that are managed over restarts for processing items (e.g. issues, timeline, ...)
+ */
+abstract class Grabber<T : Any> {
+
+    /**
+     * The response of a single step
+     */
+    interface StepResponse<T : Any> {
```

Comment:
not sure, but doesn't this generic parameter shadow the one from the outer class?

Reply:
the generic of the class is not available in the interface

Reply:
a possible solution would be to make the interface an inner abstract class
but I don't think that's worth